x-pack/packages/kbn-elastic-assistant/impl/assistant/index.tsx

```diff
@@ -251,13 +274,15 @@ const AssistantComponent: React.FC<Props> = ({
           replacements: newReplacements,
         });
 
+      const systemPrompt = allSystemPrompts.find((prompt) => prompt.id === editingSystemPromptId);
```

Comment:
This is where the system prompt selected by the user (in the editor) is added to the start of the conversation.